src/mon/MDSMonitor.cc

```diff
@@ -840,6 +850,8 @@ bool MDSMonitor::preprocess_command(MonOpRequestRef op)
   bufferlist rdata;
   stringstream ss, ds;
 
+  const auto &fsmap = get_working_fsmap();
```

Comment:
again, we are potentially replying to the client/user based on uncommitted state. And this gets a bit more annoying on the preprocess function, because we know that the client may contact the leader (and get what is likely going to turn out the stable state, but yet to be committed), or it may contact a peon and get something that is the current state. There's a short window of time when the client can get these two states from different monitors. Or it may get its state from the leader, just before the leader crashes without even starting a proposal on that value.

You know, this is uncommitted state at this point, and we are replying to the user based on it.